subsys/mgmt/settings/src/settings_mgmt.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include <mgmt/mgmt.h>
+#include <mgmt/settings_mgmt.h>
+#include <cborattr/cborattr.h>
+#include <stats/stats.h>
+
+LOG_MODULE_REGISTER(mgmt_settings, CONFIG_MGMT_SETTINGS_LOG_LEVEL);
+
+#define SETTINGS_MGMT_ID_ERASE 6
+
+static int settings_erase(struct mgmt_ctxt *ctxt)
+{
+	int rc = -EFAULT;
+	CborError cbor_err = 0;
+
+#ifdef PM_SETTINGS_STORAGE_ID
+	const struct flash_area *fa;
+	while (1) {
+		rc = flash_area_open(PM_SETTINGS_STORAGE_ID, &fa);
```

Comment:
Is it possible to use the `flash_map.h` defines here so that this would work for the `storage` partition when DT is used instead of partition manager? Like is done here https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/settings/src/settings_fcb.c#L43

Reply:
I'll check but I do think so and that's a better solution, I'll then generalize this to storage_mgmt instead of purely settings as it would be universal for any storage partition.